src/PeriodicTaskRunnerBackgroundService.cs

```diff
@@ -48,22 +48,22 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             // Ensure that we can crate the service. Do this synchronously so that we'll fail-fast no matter the failure mode
             // if the task can't run at all.
-            using (var scope = this.serviceProvider.CreateScope())
+            if(!this.taskFactory.CanResolvePeriodicTask())
             {
-                _ = scope.ServiceProvider.GetRequiredService<TPeriodicTask>();
+                this.logger.LogCritical("{Type} didn't registered in container", typeof(TPeriodicTask));
+                throw new InvalidOperationException($"{typeof(TPeriodicTask)} didn't registered in container");
             }
 
             while (!stoppingToken.IsCancellationRequested)
             {
                 try
                 {
-                    using var scope = this.serviceProvider.CreateScope();
-                    var periodicTask = scope.ServiceProvider.GetRequiredService<TPeriodicTask>();
+                    var periodicTask = this.taskFactory.GetPeriodicTask();
```

Comment:
I'm interested to hear your thoughts about what the factory does and why it's relevant - I'm not opposed to it, but I'm also not convinced of the merits.

Reply:
- New functional - user can override standart behavior
- Cleaner code